app/services/usps_in_person_proofer.rb

```diff
@@ -0,0 +1,190 @@
+class UspsInPersonProofer
+  attr_reader :token, :token_expires_at
+
+  # Makes a request to retrieve a new OAuth token
+  # and modifies self to store the token and when
+  # it expires (15 minutes).
+  # @return [String] the token
+  def retrieve_token!
+    body = request_token
+    @token_expires_at = Time.zone.now + body['expires_in']
+    @token = "#{body['token_type']} #{body['access_token']}"
+  end
+
+  def token_expired?
+    @token_expires_at.nil? || @token_expires_at.past?
+  end
+
+  # Makes HTTP request to authentication endpoint
+  # and modifies self to store the token and when
+  # it expires (15 minutes).
+  # @return [Hash] API response
+  def request_token
+    url = "#{root_url}/oauth/authenticate"
+    body = {
+      username: AppConfig.env.usps_ipp_username,
+      password: AppConfig.env.usps_ipp_password,
+      grant_type: 'implicit',
+      response_type: 'token',
+      client_id: '424ada78-62ae-4c53-8e3a-0b737708a9db',
```

Comment:
can we make this a config as well?

Reply:
I don't think it changes as I understand it. Any IPP API will use that client_id.